hw/drivers/sensors/lis2dw12/src/lis2dw12.c

```diff
@@ -1750,9 +1852,15 @@ lis2dw12_sensor_unset_notification(struct sensor *sensor, sensor_event_type_t ty
     }
     
     lis2dw12 = (struct lis2dw12 *)SENSOR_GET_DEVICE(sensor);
-    
+    itf = SENSOR_GET_ITF(sensor);
+
     lis2dw12->pdd.notify_ctx.snec_evtype &= ~type;
     lis2dw12->pdd.registered_mask &= ~LIS2DW12_NOTIFY_MASK;
+
+    rc = lis2dw12_set_wake_up_ths(itf, lis2dw12->cfg.wake_up_ths);
```

Comment:
Lets mask using LIS2DW12_WAKE_THS_THS

Reply:
Masking in the function call is probably the right way to do it.